src/Tasks/Microsoft.Common.CurrentVersion.targets

```
@@ -968,7 +968,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
   <Target Name="BuildGenerateSources" DependsOnTargets="BuildGenerateSourcesTraverse;$(BuildGenerateSourcesAction)" />
 
-  <Target Name="BuildGenerateSourcesTraverse" DependsOnTargets="PrepareProjectReferences">
+  <Target Name="BuildGenerateSourcesTraverse" DependsOnTargets="PrepareProjectReferences" Condition="'$(BuildPassReferences)' == 'true'">
```

```suggestion
  <Target Name="BuildGenerateSourcesTraverse" DependsOnTargets="PrepareProjectReferences" Condition="'$(BuildPassReferences)' == 'true' and '@(ProjectReferenceWithConfiguration)' != '' and '@(_MSBuildProjectReferenceExistent)' != '' and '%(_MSBuildProjectReferenceExistent.BuildReference)' == 'true'">
```

If we do this we can remove the condition on the task

---

No, those properties can change after resolving DependsOnTargets.

---

BuildPassReferences cannot change from evaluating DependsOnTargets? Also, this seems like a minimal perf improvement since targets don't run more than once anyway.

---

> BuildPassReferences cannot change from evaluating DependsOnTargets?
Yet.  I plan to have it.

If the condition is evaluated false, the target doesn't run but isn't "skipped" either.  It can run "again" should the condition evaluates to true later on the build and dependency is required.  This logic is used when some files are generated as during part of the build.